File: rust_study/rust_codeforces/cpp/luogu/old/P1152.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P1152.cpp
 *@data      2024/01/17 21:41:03
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

void solve() {
    int n;
    bool flag = true;
    cin >> n;
    vector<int>qwq(n);
    vector<int>ans;
    for(auto &x : qwq)cin >> x;
    for(int i = 1;i < n;++i)ans.emplace_back(abs(qwq[i] - qwq[i - 1]));
    sort(ans.begin(), ans.end());
    for(int i = 1;i < n;++i) {
        if(i != ans[i - 1]) {
            flag = false;
            break;
        }
    }
    if(flag)cout << "Jolly";
    else cout << "Not jolly";
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/old/p1152.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1152.rs
//@data      2024/01/20 15:16:38
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let qwq = cin()
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<i32>>();
    let mut qvq = (2..qwq.len())
        .map(|x| (qwq[x] - qwq[x - 1]).abs())
        .collect::<Vec<_>>();
    qvq.sort_unstable();
    if Vec::from_iter(1..=qvq.len() as i32).eq(&qvq) {
        println!("Jolly")
    } else {
        println!("Not jolly")
    }
}
